src/functional/ffront/ast_passes/simple_assign.py

```diff
@@ -67,17 +67,35 @@ class SingleAssignTargetPass(NodeYielder):
     ...     )
     ... ))
     def foo():
-        a = 1
-        b = 1
+        __sat_tmp0 = 1
+        a = __sat_tmp0
+        b = __sat_tmp0
         return (a, b)
 
     """
 
+    unique_name_id: int = 0
+
+    def _unique_symbol_name(self) -> ast.Name:
+        name = ast.Name(id=f"__sat_tmp{self.unique_name_id}", ctx=ast.Store())
+        self.unique_name_id += 1
+        return name
+
     def visit_Assign(self, node: ast.Assign) -> Iterator[ast.Assign]:
-        for target in node.targets:
-            new_assign = copy.copy(node)
-            new_assign.targets = [target]
-            yield new_assign
+        if len(node.targets) > 1:
+            synthetic_target = self._unique_symbol_name()
+            ast.copy_location(synthetic_target, node)
+            synthetic_assign = copy.copy(node)
+            synthetic_assign.targets = [synthetic_target]
+            yield synthetic_assign
+
+            for target in node.targets:
+                new_assign = copy.copy(node)
+                new_assign.targets = [target]
+                new_assign.value = ast.Name(id=synthetic_target.id, ctx=ast.Load())
+                yield new_assign
+        else:
+            yield node
```

Comment:
```suggestion
        if len(node.targets) == 1:
            yield node
            return

        synthetic_target = self._unique_symbol_name()
        ast.copy_location(synthetic_target, node)
        synthetic_assign = copy.copy(node)
        synthetic_assign.targets = [synthetic_target]
        yield synthetic_assign

        for target in node.targets:
            new_assign = copy.copy(node)
            new_assign.targets = [target]
            new_assign.value = ast.Name(id=synthetic_target.id, ctx=ast.Load())
            yield new_assign
```

I guess this is more a matter of taste, but I prefer to first read about the special cases and have the general/more complicated case later.